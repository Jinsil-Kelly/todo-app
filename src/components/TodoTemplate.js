import React from 'react';

const TodoTemplate = ({ children }) => {
  return (
    <div className="TodoTemplate" data-testid="TodoTemplate">
      <div className="app-title">TODO LIST</div>
      <div className="content">{children}</div>
    </div>
  );
};

export default TodoTemplate;
