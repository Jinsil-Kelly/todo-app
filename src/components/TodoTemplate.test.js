import React from 'react'
import { render, cleanup } from '@testing-library/react'
import TodoTemplate from './TodoTemplate'

afterEach(cleanup)

describe('<TodoTemplate />', () => {
  it('matches snapshot', () => {
    const utils = render(<TodoTemplate children="hohoho" />);
    expect(utils.container).toMatchSnapshot();
  });
  it('shows the props correctly', () => {
    const utils = render(<TodoTemplate children="hohoho" />);
    utils.getByText('hohoho'); // velopert 라는 텍스트를 가진 엘리먼트가 있는지 확인
  });
});