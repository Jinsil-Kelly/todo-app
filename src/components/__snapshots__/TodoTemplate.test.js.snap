// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoTemplate /> matches snapshot 1`] = `
<div>
  <div
    class="TodoTemplate"
  >
    <div
      class="app-title"
    >
      일정 관리
    </div>
    <div
      class="content"
    >
      hohoho
    </div>
  </div>
</div>
`;
